### PR TITLE
Fix grenade casings blocking bullets without being deleted

### LINKED ISSUE #43503: Holding an empty grenade casing allows you to block bullets without losing the casing.
Step 1. get an empty grenade casing
Step 2. hold it
Step 3. congrats you have rng chance to get hit and not take damage

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

Fixes #43503

Holding grenades had a 15% to block bullets and certain grenades would never get deleted if they didn't detonate. (like grenade casings)  Now bullets properly delete the grenade.

## Why It's Good For The Game

One more exploit removed.

## Changelog

:cl:
fix: Fix all types of grenades to be deleted when shot bullets.
/:cl:

### CODE REVIEW

**Comment by TemporalOroboros** [comment]
Can you change it to if it doesn't explode it doesn't block the projectile? That way the wand of nothing doesn't become the wand of deleting held grenades.

**Comment by timothymtorres** [comment]
I don't believe there is a proc to check if a grenade is ready to detonate.  Or if you have chems inside it that don't react and it's essentially a dud grenade.

Also a wand of nothing is an extreme edge case that you'd only see once every thousand rounds, if that.

**Comment by MrMelbert** [comment]
Why can non-explosive grenades block bullets anyways, that just seems cheesy 

If it's only a problem with unsecured chemical grenades, why not solve it there?

**Comment by timothymtorres** [comment]
I believe the original intent wasn't to block bullets but to cause the grenade to self detonate since it got "shot" by a bullet.  The self detonation should have deleted the grenade but it doesn't happen when the grenade is a casing or a dud.  (Reagents that don't combine)

So now if a grenade is shot, it gets deleted regardless.

We could also make the projectile penetrate the grenade and hit the mob and remove the return that is triggering the blocking.

**Comment by MrMelbert** [comment]
Eh, it's probably fine

File: code/game/objects/items/grenades/grenade.dm [hunk 1]
```diff
@@ -189,6 +189,9 @@
 		log_game("A projectile ([hitby]) detonated a grenade held by [key_name(owner)] at [COORD(source_turf)]")
 		message_admins("A projectile ([hitby]) detonated a grenade held by [key_name_admin(owner)] at [ADMIN_COORDJMP(source_turf)]")
 		detonate()
+		
+		if(src && !QDELETED(src)) // some grenades don't detonate but we want them destroyed
```
Comment (MrMelbert) [inline code/game/objects/items/grenades/grenade.dm:193]:
```suggestion
		if(!QDELETED(src)) // some grenades don't detonate but we want them destroyed
```
`QDELETED(src)` already implies `!src`, so you don't need to check for `src`

Comment (timothymtorres) [reply code/game/objects/items/grenades/grenade.dm:193]: We are checking two things here:

`src` is to check if the grenade is a dud and didn't detonate and `!QDELETED(src)` to make sure it wasn't garbage collected when it detonated.  Not sure if `!QDELETED` does both of these already?  `QDELETED` means an object is in the queue to get deleted if I'm not mistaken?

Comment (MrMelbert) [reply code/game/objects/items/grenades/grenade.dm:193]: Look at the macro for `QDELTED` -
`QDELTED(X) = !X || QDELING(X)`
`!QDELETED` says "do we exist or have we not qdeleted"

**Pull request merged**